AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -559,22 +613,28 @@ am_Error_e CAmSocketHandler::addTimer(const timespec & timeouts, std::function<v
         return err;
     }
 
-    static auto actionPoll = [](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
+    auto actionPoll = [this](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
     {
         uint64_t mExpirations;
-        if (read(pollfd.fd, &mExpirations, sizeof(uint64_t)) == -1)
+        if(read(pollfd.fd, &mExpirations, sizeof(uint64_t))!=sizeof(uint64_t))
         {
-            //error received...try again
-            read(pollfd.fd, &mExpirations, sizeof(uint64_t));
```

Comment:
BUG: How about the EAGAIN case? Before the idea was that the issue might be recoverable.
In fact, in case the returned ssize_t is not the one expected we should better check the errno value.
But with the current implementation AM might now throw an exception even if there is only a retry needed.